### PR TITLE
Add restart tasks script.

### PR DESCRIPTION
**Pull request by La0**
Replaces the `covdir_gen` script.
It takes a list of existing code coverage task id, uses their revision data.

I had to patch the taskcluster hookd definition to support setting the task name & group id when specified in the trigger payload.

### CODE REVIEW

File: bot/taskcluster-hook.json [hunk 1]
```diff
@@ -6,82 +6,101 @@
...
+                    "notify:email:*",
+                    "docker-worker:cache:code-coverage-bot-CHANNEL",
+                    "index:insert-task:project.releng.services.project.CHANNEL.code_coverage_bot.*"
+                ],
+                "tags": {},
+                "workerType": "releng-svc-memory"
+            }
+        ]
     },
+    "schedule": [
```
Comment (marco-c) [inline bot/taskcluster-hook.json]: Nit: why did you move the schedule?

File: bot/tools/covdir_gen.py [hunk 2]
```diff
@@ -1,132 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (marco-c) [inline bot/tools/covdir_gen.py]: We could keep this, but replace codecov.io with pushes from HGMO

File: bot/taskcluster-hook.json [hunk 3]
```diff
@@ -6,82 +6,101 @@
         "name": "Code coverage hook (CHANNEL)",
```
Comment (marco-c) [inline bot/taskcluster-hook.json]: It'd be good to have something like https://github.com/mozilla/bugbug/blob/master/tests/test_hooks.py to validate the hook definitions.

File: bot/taskcluster-hook.json (hunk 1 above)
Comment (La0) [reply bot/taskcluster-hook.json]: I did not, it's just that the alphabetical order puts it after "$merge" now

File: bot/tools/covdir_gen.py (hunk 2 above)
Comment (La0) [reply bot/tools/covdir_gen.py]: Ok, i'll cancel the removal for now

File: bot/tools/restart_tasks.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+def trigger_task(task_group_id, commit):
+    '''
+    Trigger a code coverage task to build covdir at a specified revision
+    '''
+    assert isinstance(commit, str)
+    name = 'covdir {} - {}'.format(secrets[secrets.APP_CHANNEL], commit)
+    hooks = taskcluster_config.get_service('hooks')
+    payload = {
+        'REPOSITORY': MC_REPO,
```
Comment (marco-c) [inline bot/tools/restart_tasks.py]: Could we read the repo too from the failing task? This way we support retriggering both MC and try tasks

File: bot/tools/covdir_gen.py (hunk 2 above)
Comment (marco-c) [reply bot/tools/covdir_gen.py]: And let's file an issue so we don't forget. A script like this could actually also be a superset of the 'retrigger' one.

Comment (marco-c) [reply bot/tools/covdir_gen.py]: (as it will run the bot for all revisions for which we have no coverage yet).

Comment (La0) [reply bot/tools/covdir_gen.py]: https://github.com/mozilla/code-coverage/issues/67

File: bot/taskcluster-hook.json (hunk 3 above)
Comment (La0) [reply bot/taskcluster-hook.json]: Agreed https://github.com/mozilla/code-coverage/issues/68

File: bot/tools/restart_tasks.py (hunk 4 above)
Comment (La0) [reply bot/tools/restart_tasks.py]: Done

File: bot/tools/restart_tasks.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    parser.add_argument('--group', type=str, default=slugId(), help='Task group to create/update')
+    parser.add_argument('--dry-run', action='store_true', default=False, help='List actions without triggering any new task')
+    parser.add_argument('tasks', nargs='+', help='Existing tasks to retrigger')
+    args = parser.parse_args()
+
+    # List existing tags & commits
+    print('Group', args.group)
+    try:
+        group = queue.listTaskGroup(args.group)
+        commits = [
```
Comment (marco-c) [inline bot/tools/restart_tasks.py]: Nit: a set could be better for this, since you're checking for presence at line 80
